### PR TITLE
crimson: bump up seastar version to freshest master

### PR DESCRIPTION
**Pull request by rzarzynski**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/common/errorator.h [hunk 1]
```diff
@@ -352,6 +352,14 @@ struct errorator {
     template <class ErrorVisitor, class Futurator>
     friend class maybe_handle_error_t;
 
+    // any `seastar::futurize` specialization must be able to access the base.
+    // see : `satisfy_with_result_of()` far below.
+    template <typename>
+    friend class seastar::futurize;
```
Comment (rzarzynski) [inline src/crimson/common/errorator.h:358]: No need to `plainify()` an errorated `future` nor touch the `seastar::future`. I think I missed that initially.

**Comment by rzarzynski** [comment]
Screening for FTBFSes: https://shaman.ceph.com/repos/ceph/wip-crimson-bump-up-seastar/27b9baae7c968d50e8ed9bfb922114dfddcb9e61/.

**Comment by tchaikov** [comment]
#36529

**Comment by tchaikov** [comment]
adding DNM, as it depends on #36529

**Comment by tchaikov** [comment]
currenly, seastar branch used in this changeset contains a commit proposed by https://groups.google.com/g/seastar-dev/c/kDqamdQX_28 . we should pickup the seastar upstream master after the proposed change gets merged.

**Pull request merged**